### ISSUE TITLE
[dataio] test dependencies are incomplete (Trac #1589)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1581]
parallel testing (`ctest -jX`) breaks with a clean test directory

Migrated from https://code.icecube.wisc.edu/ticket/1589
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:23", 
    "description": "parallel testing (`ctest -jX`) breaks with a clean test directory", 
    "reporter": "nega", 
    "cc": "olivas", 
    "resolution": "fixed", 
    "_ts": "1550067323910946", 
    "component": "combo core", 
    "summary": "[dataio] test dependencies are incomplete", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-03-16T06:40:05", 
    "milestone": "", 
    "owner": "jbraun", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160422T14:05:55: 
* nega commented:

> this happens on zinc. @hdembinski has seen this too, in ..... steamshovel?


* nega changed cc from "" to "hdembinski"

**Comment by icecube-trac** [comment]
Trac update at 20160422T18:32:55: 
* hdembinski changed _comment0 from:

> yes (using slack forensics...) these were the tests that failed for me (deterministic result over several attempts):
> 
>     311 - photonics-service::photonics_hitmaker_test.py (Failed)
>     312 - photonics-service::photospline_hitmaker_test.py (Failed)
>     348 - steamshovel::test_shovelio_against_I3Reader.py (Failed)
>     386 - trigger-sim::test_examples.py (Failed)

to:

> 1461350055398610


* hdembinski commented:

> yes (using slack forensics...) these were the tests that failed for me (deterministic result over several attempts):
> 
>     311 - photonics-service::photonics_hitmaker_test.py (Failed)
>     312 - photonics-service::photospline_hitmaker_test.py (Failed)
>     348 - steamshovel::test_shovelio_against_I3Reader.py (Failed)
>     386 - trigger-sim::test_examples.py (Failed)
> 
> i have a mac with el capitan.

**Comment by icecube-trac** [comment]
Trac update at 20160425T18:01:46: olivas commented:

> Please set a milestone.

**Comment by icecube-trac** [comment]
Trac update at 20160502T14:54:33: olivas commented:

> I couldn't reproduce the trigger-sim test failure and that test has no dependencies.  There are two tests that write to the same file.  This is likely the problem.  Hopefully resolved with r145257.  More information about the failure would be helpful though.

**Comment by icecube-trac** [comment]
Trac update at 20160502T15:31:02: olivas commented:

> http://code.icecube.wisc.edu/projects/icecube/changeset/145260/IceCube
> 
> That was more likely the problem.  Independent tests in sim-services and trigger-sim attempting to write to the same file.

**Comment by icecube-trac** [comment]
Trac update at 20160510T13:10:58: olivas changed milestone from "" to "Long-Term Future"

**Comment by icecube-trac** [comment]
Trac update at 20160819T18:04:53: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "nega" to "olivas"
* olivas changed milestone from "Long-Term Future" to "September 2016"
* olivas changed summary from "dataio - test dependencies are incomplete" to "[dataio] test dependencies are incomplete"

**Comment by icecube-trac** [comment]
Trac update at 20170330T18:57:52: olivas changed owner from "olivas" to "david.schultz"

**Comment by icecube-trac** [comment]
Trac update at 20170403T19:42:05: 
* david.schultz changed milestone from "September 2016" to "June 2017"
* david.schultz changed cc from "hdembinski" to ""

**Comment by icecube-trac** [comment]
Trac update at 20170522T18:28:51: 
* david.schultz commented:

> I think the answer is to just rewrite a lot of the dataio tests so they don't have any dependencies on other tests.
> 
> This sounds like a great thing for a code sprint, to just hand 5 tests to each person to rewrite.


* david.schultz changed cc from "" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20170627T13:26:04: olivas changed milestone from "June 2017" to "July 2017"

**Comment by icecube-trac** [comment]
Trac update at 20170816T20:22:18: david.schultz changed milestone from "July 2017" to "November 2017"

**Comment by icecube-trac** [comment]
Trac update at 20180607T16:14:47: olivas changed milestone from "November 2017" to "Autumnal Equinox 2018"

**Comment by icecube-trac** [comment]
Trac update at 20180922T10:31:36: olivas changed owner from "david.schultz" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20181001T21:25:29: 
* olivas commented:

> We'll keep kicking this can down the road, seeing as how this milestone date was 10 days ago.


* olivas changed milestone from "Autumnal Equinox 2018" to "Winter Solstice 2018"

**Comment by icecube-trac** [comment]
Trac update at 20181114T14:41:43: olivas changed owner from "olivas" to ""

**Comment by icecube-trac** [comment]
Trac update at 20181130T19:15:24: jbraun changed owner from "" to "jbraun"

**Comment by icecube-trac** [comment]
Trac update at 20181204T21:04:20: 
* jbraun changed status from "assigned" to "closed"
* jbraun changed resolution from "" to "fixed"
* jbraun commented:

> In changeset 3112:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="3112"
> Set test dependencies in CMakeLists.txt instead of relying on alphabetical order.  The clean solution would be to make all tests independent, but issue #2208 precludes this at them moment.  Closes #1581.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190111T22:02:46: nwhitehorn commented:

> In changeset 169267:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="169267"
> Set test dependencies in CMakeLists.txt instead of relying on alphabetical order.  The clean solution would be to make all tests independent, but issue #2208 precludes this at them moment.  Closes #1581.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:23: 
* olivas changed milestone from "Winter Solstice 2018" to ""
* olivas commented:

> Milestone Winter Solstice 2018 deleted
